src/samplers/SliceSampler.jl

```diff
@@ -110,23 +137,25 @@ function slice_shrink(h::SliceSampler, state, z, L, R, pointer, log_potential, r
     Rbar = R
 
     while true
-        U = rand(rng)
-        new_position = Lbar + U * (Rbar - Lbar)
+        new_position = draw_new_position(Lbar, Rbar, rng, typeof(pointer[]))
         pointer[] = new_position 
-        consider = (z < log_potential(state))
+        consider = z < log_potential(state)
         pointer[] = old_position
-        if (consider) && (slice_accept(h, state, new_position, z, L, R, pointer, log_potential))
+        if consider && slice_accept(h, state, new_position, z, L, R, pointer, log_potential)
             return new_position
         end
         if new_position < pointer[]
-            Lbar = new_position
+            Lbar = new_position # TODO: why does Alex have "+1" here for the IntSliceSampler?
```

Comment:
Haven't looked in detail, but my guess is that it's because Java is 0-indexed and has convention that intervals are specified in a half closed fashion, i.e. [a, b)